node/silkworm/db/mdbx.cpp

```diff
@@ -160,6 +163,37 @@ ::mdbx::cursor_managed open_cursor(::mdbx::txn& tx, const MapConfig& config) {
     return tx.open_cursor(open_map(tx, config));
 }
 
+PooledCursor::PooledCursor(::mdbx::txn& tx, const MapConfig& config) {
+    cursor_ = cursors_pool_.acquire();
+    if (!cursor_) {
+        cursor_ = std::make_unique<::mdbx::cursor_managed>();
+    }
+    bind(tx, config);
+}
+
+PooledCursor::~PooledCursor() {
+    if (cursor_->operator bool()) {
```

Comment:
why not `if (cursor_)`

Reply:
Well `if (cursor_)` returns whether the unique_pointer has a value.
Propably you meant `if(*cursor_)` which is basically a more concise syntax of what I wrote: I find it more readable in the sense that invokes the boolean operator on unique_ptr held value

Reply:
I don't think so. `if(*cursor_)` would crash is the unique_ptr was not set.
`if (cursor_)` converst the cursor to bool (using cursor_->operator bool()) and test that boolean, so it is exactly equivalent to what you wrote.

Reply:
Hmm ... Operator bool on a unique otr checks if there's an associated managed object. And in pooled cursor there always is. What I need instead is to reach the bool operator of the managed object.

Reply:
Oh, you are right, I read too fast and not noticed that you used `cursor_->` and not `cursor_.`.
So yes of course I think `if (*cursor_)` is much more readable.

Reply:
I concur with Greg: `if (*cursor_)` is more readable.